### ISSUE TITLE
marian-server throws unhelpful error without vocabularies

### ISSUE THREAD

**Issue by kpu** [marian-nmt/marian-dev#134]
Run `marian-server` without vocabularies and this comes out:
```
terminate called after throwing an instance of 'YAML::InvalidNode'
  what():  invalid node; this may result from using a map iterator as a sequence iterator, or vice-versa
```
That happens in `/mt/marian-dev/src/translator/translator.h:125`

In contrast, `marian-decoder` prints a helpful message from `/mt/marian-dev/src/data/corpus.cpp`
```
Translating but vocabularies are missing!
```

Why doesn't `command/marian_server.cpp` pass the validate argument?
```
  auto options = New<Config>(argc, argv, ConfigMode::translating);
```

**Comment by emjotde** [comment]
Hijacking this, `marian` without any options is similarly tight lipped:

```
./marian
Error: No train sets given in config file or on command line - aborting
Aborted from void marian::ConfigParser::validateOptions() const in /home/marcinj/Badania/marian/src/marian/src/common/config_parser.cpp: 142
Przerwane (core dumped)
```

Should list usage/options. I believe this was the case once.

**Comment by snukky** [comment]
Fixed both issues with ad47d7cada566fb4c22cf52c4a5aaf8c02d4f483 and e978d41f7829bd45696a6642fd3ce97f36632434.
